api/nakadi-event-bus-api.yaml

```diff
@@ -272,18 +232,6 @@ paths:
           description: Client is not authenticated
           schema:
             $ref: '#/definitions/Problem'
-        '404':
```

Comment:
> Why do you consider 404 not worth mentioning?

@rcillo, excellent question. If we consider the definition to be documentation _about the API_, then 404 says very little about things to inform _an API user_, ie, it's unremarkable that asking for an event type that does not exist will produce 404 (unless the API doesn't, which is a different thing). So on that front all it really does is make the documentation longer than it needs to be, by describing standard HTTP behaviour. 

In general, one can make the same why argument for almost all the http codes but that's going to make for a very long API document :) Opinions can vary on what significant is, but common client/server that appear in reality and are really about HTTP itself, are probably best skipped (400, 404-406, 411-418, 500, 502, 503, 504). As a counter-example I'd probably document `410 Gone` as that's not always seen in the wild (eg it might say something to an API user that this thing did once or recently exist for certain delete scenarios and give the API designer an opportunity to describe that). 

On the other hand, you'd include the very mundane 200 because it allows you to describe the response, also 201/202 as they describe how creates/updates work which tend to matter to API users, and you might include ones like 409, 422, 429 as they might say something interesting about the how the API handles edits, etc. For auth I'd generally include `401` as it lets the designer say something about how to authorize with the server (oauth, basic, api keys, etc), and I'd skip `403` unless there was a specific API scenario in mind that need to make a distinction between not allowed and won't ever allow. The `3xx` redirect range are often worth documenting _for APIs_ because it's not unusual for programmatic clients to not follow redirections (eg a media API might want to document it use and support of redirection) whereas it's built into general purpose clients like browsers.

But like I say, opinions vary on what matters :)